src/mscorlib/src/System/Guid.cs

```diff
@@ -921,9 +921,12 @@ public override String ToString()
             return ToString("D",null);
         }
 
-        public override int GetHashCode()
+        [System.Security.SecuritySafeCritical]
+        public unsafe override int GetHashCode()
         {
-            return _a ^ (((int)_b << 16) | (int)(ushort)_c) ^ (((int)_f << 24) | _k);
+            // Simply XOR all the bits of the GUID 32 bits at a time.
+            fixed (int* ptr = &this._a)
```

Comment:
Would it be better to make a copy of the struct instead of using `fixed`? e.g.

```cs
Guid self = this;
int* addr = &self._a;
return addr[0] ^ addr[1] ^ addr[2] ^ addr[3];
```

I found while working on a previous PR that the jit would not inline methods (even short ones) with `fixed`, so maybe this will help the method inline. Additionally, it could theoretically remove the additional copy if the Guid is not a member of a class, although that may face complications since we take the copy's address.